cave/plot/parallel_coordinates.py

```diff
@@ -94,41 +94,43 @@ def plot_n_configs(self, num_configs, params):
             parameters to be plotted
         """
         all_configs = self.original_rh.get_all_configs()
+        if len(all_configs) < 5:
+            self.logger.info("At least five configurations necessary for parallel coordinates!")
+            return
         # Get n most run configs
         if num_configs == -1:
             num_configs = len(all_configs)
         self.logger.debug("Plotting %d configs.", min(num_configs, len(all_configs)))
 
-        for logy in [False, True]:
-            configs_to_plot = sorted(all_configs, key=lambda x: self._fun(self.epm_rh.get_cost(x), logy))
-            # What about scenarios where quality is the value to optimize? shouldn't min and max be switched then?
-            self.best_config_performance = self._fun(min([self.epm_rh.get_cost(c) for c
-                                                          in all_configs]), logy)
-            self.worst_config_performance = self._fun(max([self.epm_rh.get_cost(c) for c
-                                                           in all_configs]), logy)
-            if num_configs < len(configs_to_plot):
-                ids = list(sorted(random.sample(range(len(configs_to_plot)), num_configs)))
-                ids[0] = 0
-                ids[-1] = len(configs_to_plot) - 1
-            else:
-                ids = list(range(len(configs_to_plot)))
-            ids[0:5] = list(range(0, 5))
-            ids[-5:] = list(range(len(configs_to_plot) - 6, len(configs_to_plot) - 1))
-            self._plot(np.array(configs_to_plot)[ids], params,
-                       fn=os.path.join(self.output_dir, "parallel_coordinates_uniform_{:s}".format(
-                           'log_cost' if logy else '') + str(len(ids)) + '.png'), logy=logy)
-            if num_configs < len(configs_to_plot):  # Only sample on the logscale if not all configs are plotted.
-                ids = self._get_log_spaced_ids(configs_to_plot, num_configs)
-            else:
-                ids = list(range(len(all_configs)))
-            ids[0:5] = list(range(0, 5))
-            ids[-5:] = list(range(len(configs_to_plot) - 6, len(configs_to_plot) - 1))
-            configs_to_plot = np.array(configs_to_plot)[ids]
-            res = self._plot(configs_to_plot, params,
-                             fn=os.path.join(self.output_dir, "parallel_coordinates_{:s}".format(
-                                 'log_cost' if logy else ''
-                             ) + str(len(ids)) + '.png'), logy=logy)
-        return res
+        config_to_cost = {c : self.epm_rh.get_cost(c) for c in all_configs}
+        pngs = []
+        for log_cost in [False, True]:
+            for log_sample in [False, True]:
```

Comment:
I don't see why we need the extra for here.
it could be user specified if log_cost should be used and if no input was specified put it on log if the metric is runtime.